src/DataLib/ALARAJOY.cpp

```diff
@@ -0,0 +1,181 @@
+#include "ALARAJOY.h"
+#include "DataLib/ALARALib_def.h"
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <string>
+#include <cstring>
+#include <cstdio>
+
+// Define static member
+std::vector<CSVRow> ALARAJOYLib::csvData;
+
+// Free helper: Parse cross section array from Python list (of floats) format
+static std::vector<float> parseXSectionArray(const std::string& arrayStr) {
+    std::vector<float> xs;
+    if (arrayStr.size() < 2) return xs;
+
+    std::string clean = arrayStr.substr(1, arrayStr.size()-2);
+    std::stringstream ss(clean);
+    std::string item;
+
+    while (std::getline(ss, item, ',')) {
+        size_t a = item.find_first_not_of(" \t");
+        size_t b = item.find_last_not_of(" \t");
+        if (a == std::string::npos) continue;
+        xs.push_back(std::stof(item.substr(a, b - a + 1)));
+    }
+    return xs;
+}
+
+// Free helper: Parse a single CSV row
+static CSVRow parseCSVRow(const std::string& line)
```

Comment:
This should be a member function of the ALARAJOY class.  (or maybe CSVRow is a class and this is a member function of that?)